pkg/sloop/store/typed/eventcounttable.go

```diff
@@ -76,3 +79,14 @@ func (t *ResourceEventCountsTable) GetOrDefault(txn badgerwrap.Txn, key string)
 func (k *EventCountKey) SetPartitionId(newPartitionId string) {
 	k.PartitionId = newPartitionId
 }
+
+func (k *EventCountKey) GetKeyPrefixString() string {
```

Comment:
Lets hold on this for now, and go back to use string

Reply:
For the purpose of faster event lookup, we only need a tweak to the key for watch table.  And one extra if will do:

```
func (k *WatchTableKey) String() string {
	if k.Name == "" && k.Timestamp.IsZero() {
		return fmt.Sprintf("/%v/%v/%v/%v/", k.TableName(), k.PartitionId, k.Kind, k.Namespace)
	} else if k.Timestamp.IsZero() {
		return fmt.Sprintf("/%v/%v/%v/%v/%v/", k.TableName(), k.PartitionId, k.Kind, k.Namespace, k.Name)
	} else {
		return fmt.Sprintf("/%v/%v/%v/%v/%v/%v", k.TableName(), k.PartitionId, k.Kind, k.Namespace, k.Name, k.Timestamp.UnixNano())
	}
}
```

We can build a better key class in a future PR